src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -556,6 +574,27 @@ std::shared_ptr<ov::ICompiledModel> Plugin::compile_model(const std::shared_ptr<
         localConfig.update({{ov::intel_npu::batch_mode.name(), strStream.str()}});
     }
 
+    ov::Dimension originalBatch;
+    bool modelDeBached = false;
+    if (localConfig.isAvailable(ov::intel_npu::batch_mode.name()) &&
+        localConfig.get<BATCH_MODE>() == ov::intel_npu::BatchMode::PLUGIN && model->is_dynamic()) {
+        try {
+            originalBatch = ov::get_batch(modelForCompilation);
+            ov::set_batch(modelForCompilation, 1);
+            modelDeBached = true;
+        } catch (const std::exception& ex) {
+            _logger.warning("The plugin couldn't resize a batched model due to exception: {0}.\nProbably, the "
+                            "model is a dynamic model and layout hasn't been specified. Trying to debatch it...",
+                            ex.what());
+            modelForCompilation = deBatchModel(modelForCompilation);
+            if (!modelForCompilation) {
+                OPENVINO_THROW("Cannot debatch a model");
+            }
+            _logger.info("The model has been debatched successfully");
+            modelDeBached = true;
+        }
+    }
```

Comment:
TBD: Move it to a separate `debatchDynamicModel` function and add batch consistency checks.